appinventor/blocklyeditor/src/drawer.js

```diff
@@ -463,140 +454,47 @@ Blockly.Drawer.prototype.componentTypeToXMLArray = function(typeName) {
-        }
-        xmlString += Blockly.Drawer.mutatorAttributesToXMLString(mutatorAttributes);
-      }
-      xmlString += '</block></xml>';
-    }
-  }
-  return Blockly.Xml.textToDom(xmlString);
-}
-
-Blockly.Drawer.mutatorAttributesToXMLString = function(mutatorAttributes){
-  var xmlString = '<mutation ';
-  for(var attributeName in mutatorAttributes) {
-    if (!mutatorAttributes.hasOwnProperty(attributeName)) continue;
-    xmlString += attributeName + '="' + mutatorAttributes[attributeName] + '" ';
-  }
-  xmlString += '></mutation>';
-  return xmlString;
-};
-
-/**
- * Creates the xml for a value input with the given name.
- * @param {string} name The name of the value input.
- * @return {!Node} Two xml nodes defining the value input. The outer is an <xml>
- *     node and the inner is a <value> node.
- */
-Blockly.Drawer.prototype.valueInputToXML = function(name) {
-  var xmlString = '<xml><value name="' + name + '"></value></xml>';
-  return Blockly.Xml.textToDom(xmlString);
-}
+    var utils = Blockly.Util.xml;
 
-/**
- * @param {string} name The name of the value input.
- * @param {HelperKey} helperKey Key defining the helper block to create.
- * @return {!Node} XML nodes defining a value input with a helper block attached.
- */
-Blockly.Drawer.prototype.valueWithHelperXML = function(name, helperKey) {
-  var inputXml= this.valueInputToXML(name);
-  var helperXml= this.helperKeyToXML(helperKey);
-  // First child b/c these are wrapped in an <xml/> node.
-  inputXml.firstChild.appendChild(helperXml.firstChild);
-  return inputXml;
-}
-
-/**
- * Converts a group of nodes (the outer-most one being a <xml> node) int an
- * array of all of the child nodes.
- * @param {!Node} xml A group of nodes, the outer most one being an <xml> node.
- * @return {!Array<!Node>} An array of xml nodes.
- */
-Blockly.Drawer.prototype.XMLToArray = function(xml) {
-  var xmlArray = [];
-  // [lyn, 11/10/13] Use goog.dom.getChildren rather than .children or
-  //   .childNodes to make this code work across browsers.
-  var children = goog.dom.getChildren(xml);
-  for(var i = 0, child; child = children[i]; i++) {
-    xmlArray.push(child);
-  }
-  return xmlArray;
-}
-
-// [lyn, 10/22/13] return an XML string including one procedure caller for each procedure declaration
-// in main workspace.
-// [jos, 10/18/15] if we pass a proc_name, we only want one procedure returned as xmlString
-Blockly.Drawer.prototype.procedureCallersXMLString = function(returnsValue, proc_name) {
-  var xmlString = '<xml>';  // Used to accumulate xml for each caller
-  var decls = this.workspace_.getProcedureDatabase().getDeclarationBlocks(returnsValue);
-
-  if (proc_name) {
-    for (var i = 0; i < decls.length; i++) {
-      if (decls[i].getFieldValue('NAME').toLocaleLowerCase() == proc_name){
-        xmlString += Blockly.Drawer.procedureCallerBlockString(decls[i]);
-        break;
+    function defaultCase() {
```

Comment:
Given that this is only called once, it seems like it would be better to just have it in line with where it is used.